tests/TransactionTest.php

```diff
@@ -291,6 +291,35 @@ public function testDecrementWithRollBack(): void
         $this->assertTrue(DB::collection('users')->where('name', 'klinson')->where('age', 20)->exists());
     }
 
+    /**
+     * @testWith [false]
+     *           [true]
+     */
+    public function testCount(bool $transaction)
+    {
+        if ($transaction) {
+            DB::beginTransaction();
+        }
+
+        $this->assertEquals(0, DB::collection('users')->count());
+        $this->assertEquals(0, DB::collection('users')->where('age', 20)->count());
+        DB::collection('users')->insert(['name' => 'klinson', 'age' => 20, 'title' => 'admin']);
+        DB::collection('users')->insert(['name' => 'bryan', 'age' => 18, 'title' => 'user']);
+        $this->assertEquals(2, DB::collection('users')->count());
+
+        // Counting fetched results
+        $this->assertEquals(1, DB::collection('users')->where('age', 20)->get()->count());
+
+        if ($transaction) {
+            // until transaction is committed, count with filter is incorrect
```

Comment:
This should not be the case. Consider:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

$client = new MongoDB\Client('mongodb://localhost:27070/?replicaSet=rs0');
$collection = $client->test->foo;
$collection->drop();

$session = $client->startSession();

echo "Starting transaction\n";
$session->startTransaction();

printf("countDocuments: %d\n", $collection->countDocuments([], ['session' => $session]));

echo "Inserting two documents\n";
$collection->insertOne(['name' => 'klinson', 'age' => 20, 'title' => 'admin'], ['session' => $session]);
$collection->insertOne(['name' => 'bryan', 'age' => 18, 'title' => 'user'], ['session' => $session]);

printf("countDocuments: %d\n", $collection->countDocuments([], ['session' => $session]));

$filter = ['age' => 20];
printf("countDocuments(%s): %d\n", json_encode($filter), $collection->countDocuments($filter, ['session' => $session]));

echo "Committing transaction\n";
$session->commitTransaction();

printf("count: %d\n", $collection->count([], ['session' => $session]));
printf("countDocuments: %d\n", $collection->countDocuments([], ['session' => $session]));
printf("estimatedDocumentCount: %d\n", $collection->estimatedDocumentCount(['session' => $session]));

printf("count(%s): %d\n", json_encode($filter), $collection->count($filter, ['session' => $session]));
printf("countDocuments(%s): %d\n", json_encode($filter), $collection->countDocuments($filter, ['session' => $session]));
```

Output:

``` 
Starting transaction
countDocuments: 0
Inserting two documents
countDocuments: 2
countDocuments({"age":20}): 1
Committing transaction
count: 2
countDocuments: 2
estimatedDocumentCount: 2
count({"age":20}): 1
countDocuments({"age":20}): 1
```

Note that per [Transactions: Count Operation](https://www.mongodb.com/docs/manual/core/transactions/#count-operation), the `count` command cannot be run within a transaction, so your only option is `countDocuments()` there.

If `countDocuments` isn't getting an accurate view of the collection, that suggests the library may not be correctly associating it with the transaction (i.e. session may not be passed).